src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -515,6 +516,39 @@ await foreach (var response in requests)
             return colorInformation;
         }
 
+        public override async Task<IReadOnlyList<ColorPresentation>> ProvideHtmlColorPresentationAsync(ColorPresentationParams colorPresentationParams, CancellationToken cancellationToken)
+        {
+            if (colorPresentationParams is null)
+            {
+                throw new ArgumentNullException(nameof(colorPresentationParams));
+            }
+
+            var htmlDoc = GetHtmlDocumentSnapshot(colorPresentationParams.TextDocument.Uri);
+            if (htmlDoc is null)
+            {
+                return Array.Empty<ColorPresentation>();
+            }
+
+            colorPresentationParams.TextDocument.Uri = htmlDoc.Uri;
+            var htmlTextBuffer = htmlDoc.Snapshot.TextBuffer;
+            var requests = _requestInvoker.ReinvokeRequestOnMultipleServersAsync<ColorPresentationParams, ColorPresentation[]>(
```

Comment:
Why multiple servers? The rest of this code assumes that only HTML will support this, and does no mapping, so wouldn't it make more sense to check that and only request from the html server?

Reply:
IIRC the idea is that even if we know only HTML servers will support this there might still be multiple HTML servers registered that we ought to check all of them.